extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -293,6 +293,16 @@ export class SubscriptionService {
           entitlement =
             entitlement ||
             Entitlement.empty(subscriptionPlatform.getServiceId());
+          if (this.checkEntitlementsAndEncryptionInvalid_(entitlement)) {
```

Comment:
Can't we just wrap the whole thing in the function and return the entitlement or log an error? This still looks like most of shared code is repeated.

Reply:
I don't really like the idea of logging inside of the small helper function, especially since we would need to pass in context about the calling function (the service ID) into the helper function in order to enable logging. Also, we would still need to do a null check in the original function, so I don't think the tradeoff is very clear, i.e.
```
const ent = this.checkEntitlementsAndEncryptionInvalid_(entitlements);
if (!ent) {
  return null;
}

Reply:
Sure. But what I'm thinking about instead, is wrapping the whole `subscriptionPlatform.getEntitlements()` into a shared function. E.g.

```
/**
 * @return {!Promise<?Entitlements>}
 */
getEntitlements_(platform) {
  return platform.getEntitlements().
     then(entitlements => {
       if (entitlements &&
           entitlements.granted &&
           isDocEncrypted &&
           !entitlements.decryptedKey) {
         const logChannel = platform.getServiceId() == 'local' ? user() : dev();
         logChannel.error(TAG, `${platform.getServiceId()}: ....`);
         return null;
       }
       return entitlements;
     });
}
```

Then both calls could look very similar. They'd just use this function instead of calling `platform.getEntitlements()` directly.

Makes sense?

Reply:
Sure does. Thanks for spelling it out for me.